src/NuGet.Core/NuGet.Protocol/HttpSource/HttpSourceAuthenticationHandler.cs

```diff
@@ -198,7 +203,13 @@ private async Task<ICredentials> AcquireCredentialsAsync(HttpStatusCode statusCo
                         _packageSource.Source);
                 }
 
-                var promptCredentials = await PromptForCredentialsAsync(
+                ICredentials promptCredentials = default;
+                if (!areLastKnownGoodCredentialsTried)
+                {
+                    _ = _credentialService.TryGetLastKnownGoodCredentialsFromCache(uri: _packageSource.SourceUri, isProxy: false, out promptCredentials);
```

Comment:
I have hardcoded, `isProxy: false` because the if statement above sets the type to either `Unauthorized` or `Forbidden`. Hence it can't be [CredentialRequestType.Proxy](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Configuration/Credential/CredentialRequestType.cs#L8).

Reply:
This would be more useful as a comment, so that anyone reading the code after this PR is merged, can get the same context.

Reply:
Fixed in the latest commit.